sensor/kubernetes/eventpipeline/component/message.go

```diff
@@ -73,13 +77,47 @@ func (e *ResourceEvent) AddDeploymentForReprocessing(ids ...string) *ResourceEve
 	return e
 }
 
-// AddDeploymentReference creates and sets a new deployment reference to this resource event.
-func (e *ResourceEvent) AddDeploymentReference(reference resolver.DeploymentReference, action central.ResourceAction, force bool) {
-	e.DeploymentReferences = append(e.DeploymentReferences, DeploymentReference{
+// DeploymentReferenceOption option function for the DeploymentReference struct
+type DeploymentReferenceOption func(*DeploymentReference)
+
+// DeploymentRefWithParentResourceAction sets the ParentResourceAction
+func DeploymentRefWithParentResourceAction(action central.ResourceAction) DeploymentReferenceOption {
+	return func(dr *DeploymentReference) {
+		dr.ParentResourceAction = action
+	}
+}
+
+// DeploymentRefWithForceDetection sets whether the detection should be forced or not
+func DeploymentRefWithForceDetection(force bool) DeploymentReferenceOption {
+	return func(dr *DeploymentReference) {
+		dr.ForceDetection = force
+	}
+}
+
+// DeploymentRefWithSkipResolving sets whether the deployment should be resolved or not
+func DeploymentRefWithSkipResolving(skip bool) DeploymentReferenceOption {
```

Comment:
I think the `bool` arg opt functions will read much better without the param. It's defaulted to `false` anyway, so there's no reason to set it to `false` when calling the opt. Just do it:
```go
func DeploymentRefWithSkipResolving() DeploymentReferenceOption {
	return func(dr *DeploymentReference) {
		dr.SkipResolving = true
	}
```

Reply:
Done.